app/services/pipeline_report_service.rb

```diff
@@ -474,22 +474,23 @@ def find_taxa_to_highlight(sorted_genus_tax_ids, counts_by_tax_level)
     return highlighted_tax_ids
   end
 
-  def tag_pathogens(counts_by_tax_level, lineage_by_tax_id)
-    counts_by_tax_level[TaxonCount::TAX_LEVEL_SPECIES].each do |species_tax_id, species_info|
-      lineage = lineage_by_tax_id[species_tax_id] ? lineage_by_tax_id[species_tax_id] : lineage_by_tax_id[species_info[:genus_tax_id]]
-      name_columns = lineage.keys.select { |cn| cn.include?("_name") }
+  def tag_pathogens(counts_by_tax_level)
+    counts_by_tax_level[TaxonCount::TAX_LEVEL_SPECIES].each do |_species_tax_id, species_info|
       pathogen_tags = []
       TaxonLineage::PRIORITY_PATHOGENS.each do |category, pathogen_list|
         pathogen_tags |= [category] if pathogen_list.include?(species_info[:name])
-        genus_name = counts_by_tax_level[TaxonCount::TAX_LEVEL_GENUS][species_info[:genus_tax_id]][:name]
-        pathogen_tags |= [category] if pathogen_list.include?(genus_name)
-        name_columns.each do |col|
-          pathogen_tags |= [category] if pathogen_list.include?(lineage[col])
-        end
       end
       best_tag = pathogen_tags[0] # first element is highest-priority element (see PRIORITY_PATHOGENS documentation)
       species_info['pathogenTag'] = best_tag
     end
+    counts_by_tax_level[TaxonCount::TAX_LEVEL_GENUS].each do |_genus_tax_id, genus_info|
+      pathogen_tags = []
```

Comment:
Can we put the content of this block on a method e.g. `get_best_pathogen_tag` and reuse it for species and genus?

Reply:
Done